frontend/src/metabase/static-viz/components/XYChart/utils/ticks.ts

```diff
@@ -181,7 +181,13 @@ export function fixTimeseriesTicksExceedXTickCount(
         }
       });
 
-    return minLengthTicks;
+    if (minLengthTicks.length <= numTicks) {
+      return minLengthTicks;
+    }
+
+    return minLengthTicks.filter(
+      (_, index, ticks) => index % Math.ceil(ticks.length / numTicks) === 0,
```

Comment:
@JesseSDevaney I applied your suggestion + ensured the result ticks length doesn't go over `numTicks`

The original logic was introduced in https://github.com/airbnb/visx/pull/752, there was no explanation why this logic was like that.

I used `Math.ceil` instead to ensure that we don't go over `numTicks` i.e. if we want to chop up 5-length ticks into 4-length ticks we must filter the ticks every other ticks. Because 5/4 = 1.25 if we use `Math.round` we'll still get 1 and the original logic used `ticks.length - 1` divided by `numTicks`, I'm not totally sure why they specifically use `ticks.length - 1` instead of just `ticks.length` since that compares with the use of `Math.round` could result in the ticks length that is longer `numTicks` in many cases.

Considering the static viz which has limited space we'd only see a handful of ticks anyway, so the newly modified logic is accounted for that.

Reply:
This, however, changes some of the charts we're rendering e.g. https://percy.io/6a8dd5c9/metabase/builds/30026586/changed/1662698150?browser=chrome&browser_ids=48&subcategories=unreviewed%2Cchanges_requested&viewLayout=overlay&viewMode=new&width=1280&widths=1280

https://github.com/metabase/metabase/blob/645927ceebb0394cda4c077af07ec89c760b8a61/frontend/src/metabase/static-viz/components/XYChart/XYChart.tsx#L150

Since we hard code the `numTicks` to 4 and the previous version of the code doesn't ensure the number of ticks doesn't go over 4 we could sometimes render more than 4 ticks. With the new logic we ensure that we will not render more than 4 ticks and in some cases would make the chart look a little bit worse (see the Percy snapshot above) e.g. when we used to render 5 ticks and now only down to 3

Please let me know if we should still rework this more. I however think this is still better than having a ton of charts having overlapping x tick labels.